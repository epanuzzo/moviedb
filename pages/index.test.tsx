import React from "react";
import renderer from "react-test-renderer";
import HomePage from "./index";
import { useDispatch, useSelector } from "react-redux";

jest.mock("react-redux", () => ({
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));

jest.mock("../redux/slices/movieSlice", () => ({
  getMovies: jest.fn(),
}));

describe("HomePage", () => {
  let mockDispatch: jest.Mock;

  beforeEach(() => {
    mockDispatch = jest.fn();
    (useDispatch as unknown as jest.Mock).mockReturnValue(mockDispatch);
    (useSelector as unknown as jest.Mock).mockReturnValue({
      movies: [
        { id: 1, title: "Movie 1" },
        { id: 2, title: "Movie 2" },
      ],
      loading: false,
      pages: 2,
      currentPage: 1,
    });
  });

  it("renders the component", () => {
    const tree = renderer.create(<HomePage />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("displays loading state", () => {
    (useSelector as unknown as jest.Mock).mockReturnValueOnce({
      movies: [],
      loading: true,
      pages: 0,
      currentPage: 0,
    });

    const tree = renderer.create(<HomePage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
