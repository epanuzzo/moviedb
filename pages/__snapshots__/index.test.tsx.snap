// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage displays loading state 1`] = `
<div
  className="w-full h-full"
>
  <div
    className="flex justify-center items-center h-48"
  >
    <h1
      className="text-4xl font-bold text-center"
    >
      Welcome to the best 1990 movies website
    </h1>
  </div>
  <div
    className="flex justify-center items-center pt-4"
  >
    <div
      className="animate-spin rounded-full h-16 w-16 border-t-4 border-blue-500 border-solid"
    />
  </div>
  0
</div>
`;

exports[`HomePage renders the component 1`] = `
<div
  className="w-full h-full"
>
  <div
    className="flex justify-center items-center h-48"
  >
    <h1
      className="text-4xl font-bold text-center"
    >
      Welcome to the best 1990 movies website
    </h1>
  </div>
  <div
    className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-4 p-4"
  >
    <a
      className="relative bg-white rounded-md shadow-lg overflow-hidden h-80 hover:cursor-pointer"
      href="/movie/1"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="w-full h-full bg-gray-800 flex items-center justify-center"
      >
        <p
          className="text-white"
        >
          Image not found
        </p>
      </div>
      <div
        className="absolute bottom-0 left-0 right-0 p-4 bg-gradient-to-t from-black to-transparent"
      >
        <h3
          className="text-lg font-semibold text-white"
        >
          Movie 1
        </h3>
      </div>
      <div
        className="absolute top-4 left-4 bg-white text-black rounded-full flex items-center justify-center w-10 h-10"
      >
        <p
          className="text-sm font-bold"
        />
      </div>
    </a>
    <a
      className="relative bg-white rounded-md shadow-lg overflow-hidden h-80 hover:cursor-pointer"
      href="/movie/2"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="w-full h-full bg-gray-800 flex items-center justify-center"
      >
        <p
          className="text-white"
        >
          Image not found
        </p>
      </div>
      <div
        className="absolute bottom-0 left-0 right-0 p-4 bg-gradient-to-t from-black to-transparent"
      >
        <h3
          className="text-lg font-semibold text-white"
        >
          Movie 2
        </h3>
      </div>
      <div
        className="absolute top-4 left-4 bg-white text-black rounded-full flex items-center justify-center w-10 h-10"
      >
        <p
          className="text-sm font-bold"
        />
      </div>
    </a>
  </div>
  <div
    className="flex justify-center items-center space-x-4 mt-8"
  >
    <button
      className="px-2 md:px-4 py-2 rounded-md bg-gray-700 text-white font-semibold hover:bg-gray-600 transition opacity-50 cursor-not-allowed"
      disabled={true}
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="h-6 w-6"
        fill="currentColor"
        viewBox="0 0 20 20"
      >
        <path
          clipRule="evenodd"
          d="M12.79 5.23a.75.75 0 01-.02 1.06L8.832 10l3.938 3.71a.75.75 0 11-1.04 1.08l-4.5-4.25a.75.75 0 010-1.08l4.5-4.25a.75.75 0 011.06.02z"
          fillRule="evenodd"
        />
      </svg>
    </button>
    <div
      className="flex space-x-1 md:space-x-2"
    >
      <button
        className="px-2 md:px-4 py-2 rounded-md bg-blue-600 text-white"
        onClick={[Function]}
      >
        1
      </button>
      <button
        className="px-2 md:px-4 py-2 rounded-md bg-gray-700 text-white hover:bg-gray-600"
        onClick={[Function]}
      >
        2
      </button>
    </div>
    <button
      className="px-2 md:px-4 py-2 rounded-md bg-gray-700 text-white font-semibold hover:bg-gray-600 transition "
      disabled={false}
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="h-6 w-6"
        fill="currentColor"
        viewBox="0 0 20 20"
      >
        <path
          clipRule="evenodd"
          d="M7.21 14.77a.75.75 0 01.02-1.06L11.168 10 7.23 6.29a.75.75 0 111.04-1.08l4.5 4.25a.75.75 0 010 1.08l-4.5 4.25a.75.75 0 01-1.06-.02z"
          fillRule="evenodd"
        />
      </svg>
    </button>
  </div>
</div>
`;
